server/s3_get_bucket_action.cc

```diff
@@ -214,10 +216,23 @@ void S3GetBucketAction::get_next_objects_successful() {
         // Filter out by prefix
         if (kv.first.find(request_prefix) == std::string::npos) {
           // Key does not start with specified prefix; key filetered out.
-          // Check the next key.
+          // If previous/last key was matching the prefix, stop and break
+          // further enumeration.
+          if (last_key_prefix_match) {
+            // No further prefix match will occur (as items in Motr storage are
+            // arranaged in lexical order)
+            skip_no_further_prefix_match = true;
+            // Set length to zero to indicate truncation is false
+            length = 0;
+            s3_log(
+                S3_LOG_DEBUG, request_id,
+                "No further prefix match. Skipping further object listing\n");
```

Comment:
Have request_prefix printed in log

Reply:
yes. it is in log at debug level

Reply:
do you want to print request prefix at line #227?

Reply:
Its ok